src/events/http.js

```diff
@@ -215,6 +216,69 @@ function httpWrapper(wrappedFunction) {
                 this, buildParams(url, options, patchedCallback)
             );
 
+            /**
+             * Wraps 'on' method in a response to capture data event.
+             * @param {Function} wrappedResFunction The wrapped end function
+             * @returns {Function} The wrapped function
+             */
+            function responseOnWrapper(wrappedResFunction) { // eslint-disable-line no-inner-declarations,max-len
+                return function internalResponseOnWrapper(resEvent, resCallback) {
+                    if (resEvent !== 'data' || typeof resCallback !== 'function') {
+                        return wrappedResFunction.apply(this, [resEvent, resCallback]);
+                    }
+                    const resPatchedCallback = (chunk) => {
+                        if (chunk) {
+                            const totalSize = chunks.reduce(
+                                (total, item) => item.length + total,
+                                0
+                            );
```

Comment:
don't make the calculation each time, keep a sum

Reply:
👍